### PR TITLE
Move PMI code from src/arch/ofi to src/arch/util/proc_management

### PR DESCRIPTION
**Pull request by nitbhat**
This commit moves all the PMI related code and licenses from src/arch/ofi to
src/arch/util/proc_management.

### CODE REVIEW

**Comment by evan-charmworks** [comment]
This reorganization looks good. Did we want to explore any other options for what to call the `proc_management` directory?

**Comment by nitbhat** [comment]
> This reorganization looks good. Did we want to explore any other options for what to call the `proc_management` directory?

Yes, I couldn't come up with a better name. I'm open to suggestions.

**Comment by nitbhat** [comment]
Since all of the process management is PMI, we can also rename the directory to pmi.

File: src/scripts/Makefile [hunk 1]
```diff
@@ -365,35 +365,36 @@ endif
...
-	./gathertree $(SRCBASE)/QuickThreads     QuickThreads
-	./gathertree $(SRCBASE)/../contrib/hwloc hwloc
-	./gathertree $(SRCBASE)/libs             libs
-	./gathertree $(SRCBASE)/arch/util        .
-	./gathertree $(SRCBASE)/util             .
-	./gathertree $(SRCBASE)/langs            langs
-	./gathertree $(SRCBASE)/langs/jade       langs/jade
-	./gathertree $(SRCBASE)/arch/common      .
-	./gathertree $(SRCBASE)/arch/$(CMK_GDIR) .
+	$(SRCBASE)/scripts/gatherflat $(SRCBASE)/scripts          .
```
Comment (brminich) [inline src/scripts/Makefile]: minor: this `.` is not aligned by column

**Pull request merged**